src/app/integration_test/dad_jokes_page_test.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:app/access/mocking/mocking_repository.dart';
+import 'package:app/app.dart';
+import 'package:app/app_router.dart';
+import 'package:app/main.dart';
+import 'package:app/presentation/dad_jokes/dad_joke_list_item.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:get_it/get_it.dart';
+import 'package:integration_test/integration_test.dart';
+
+Future<void> main() async {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+  await initializeComponents(isMocked: true);
+
+  tearDownAll(
+    () async => await GetIt.I.get<MockingRepository>().setMocking(false),
+  );
+
+  testWidgets('Get Dad Jokes', (WidgetTester tester) async {
+    // Arrange
+
+    // Act
+    await tester.pumpWidget(const App());
+
+    // Without this the dadjokesContainer isn't there yet.
+    await tester.pumpAndSettle();
+
+    // Assert
+    var dadjokesContainer = find.byKey(const Key('DadJokesContainer'));
+
+    var dadJokes = find.descendant(
+      of: dadjokesContainer,
+      matching: find.byType(DadJokeListItem),
+    );
+
+    expect(dadJokes, findsAtLeast(1));
+  });
+
+  group("Favorites", () {
+    testWidgets('Can remove joke from favorite', (WidgetTester tester) async {
+      // Arrange
+      await tester.pumpWidget(const App());
+      await tester.pumpAndSettle();
+
+      router.go(favoriteDadJokesPagePath);
+      await tester.pumpAndSettle();
+
+      var favJokesContainer = find.byKey(const Key('FavoriteJokesContainer'));
+      var favJokes = find.descendant(
+        of: favJokesContainer,
+        matching: find.byType(DadJokeListItem),
+      );
+
+      // Act
+      await tester.tap(favJokes.first);
+      await tester.pumpAndSettle();
```

Comment:
nitpick: to be consistent, shouldn't the following comment ` // Without this the dadjokesContainer isn't there yet.` be there?

Reply:
Not really because here it is used to trigger a frame change and where I put the comments it is used to wait for the app to be fully loaded before moving on